zerver/views/messages.py

```diff
@@ -706,6 +709,30 @@ def zcommand_backend(request: HttpRequest, user_profile: UserProfile,
         ret = dict(msg=msg)
         return json_success(ret)
 
+    if tokens[0] == 'mute_topic':
+        if len(tokens) < 3 or tokens[1] == 'help':
+            ret = dict(msg="Usage: /mute_topic <stream_name> <topic_name>.")
+            return json_success(ret)
+
+        topic = ' '.join(tokens[2:])
+        stream_name = tokens[1]
+        try:
+            stream = get_realm_stream(stream_name=stream_name,
+                                      realm_id=user_profile.realm.id)
+            recipient = get_stream_recipient(stream.id)
+            if not topic_exists(topic, recipient):
+                msg = ("A valid topic name is required.")
+            else:
+                if topic_is_muted(user_profile, stream.id, topic):
+                    msg = ("The topic '%s' is already muted." % (topic))
```

Comment:
If the user tries to mute a topic which has already been muted, should we unmute it? Slack does this.

Reply:
I think this should be the case since hotkey `M` does that.